R/pipe_continuation_linter.R

```diff
@@ -0,0 +1,41 @@
+#' @describeIn linters check that all commas are followed by spaces, but do not
+#' have spaces before them.
+#' @importFrom xml2 xml_find_all as_list
+#' @export
+pipe_continuation_linter <- function(source_file) {
+  x <- global_xml_parsed_content(source_file)
+  if (is.null(x)) {
+    return()
+  }
+
+  # Find all expressions that span more than one line
+  # and contain specials named '%>%' that are on the same line
+
+  expressions_spanning_multiple_lines <- "expr[@line1 != @line2]"
+  pipes_on_same_line <- "SPECIAL[text() = '%>%' and @line1 = following::SPECIAL[text() = '%>%']/@line1]"
+
+  exprs <- xml_find_all(x,
+    p("//",
+      expressions_spanning_multiple_lines,
+      "//",
+      pipes_on_same_line
+    ))
+
+  lapply(exprs,
+    function(expr) {
+      x <- as_list(expr)
+      line <- get_file_line(source_file, x@line1)
+      Lint(
+        filename = source_file$filename,
+        line_number = x@line1,
+        column_number = x@col2,
+        type = "style",
+        message = "Each `%>%` statement should be on it's own line.",
```

Comment:
Perhaps two different messages make sense here?
"`%>%` should be preceded by a space.",
"`%>%` should be followed by a space (if the pipe fits on a single line) or a newline (otherwise)."